internal/handlers/generate_sbom.go

```diff
@@ -0,0 +1,109 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	trivyCommands "github.com/aquasecurity/trivy/pkg/commands"
+	storagev1alpha1 "github.com/rancher/sbombastic/api/storage/v1alpha1"
+	"github.com/rancher/sbombastic/internal/messaging"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GenerateSBOMHandler struct {
+	k8sClient client.Client
+	logger    *zap.Logger
+}
+
+func NewGenerateSBOMHandler(k8sClient client.Client, logger *zap.Logger) *GenerateSBOMHandler {
+	return &GenerateSBOMHandler{
+		k8sClient: k8sClient,
+		logger:    logger.Named("generate_sbom_handler"),
+	}
+}
+
+func (h *GenerateSBOMHandler) Handle(message messaging.Message) error {
+	generateSBOMMessage, ok := message.(*messaging.GenerateSBOM)
+	if !ok {
+		return fmt.Errorf("expected GenerateSBOM, got %T", message)
+	}
+
+	h.logger.Debug("SBOM generation requested",
+		zap.String("image", generateSBOMMessage.ImageName),
+		zap.String("namespace", generateSBOMMessage.ImageNamespace),
+	)
+
+	ctx := context.Background()
+
+	image := &storagev1alpha1.Image{}
+	err := h.k8sClient.Get(ctx, client.ObjectKey{
+		Name:      generateSBOMMessage.ImageName,
+		Namespace: generateSBOMMessage.ImageNamespace,
+	}, image)
+	if err != nil {
+		return fmt.Errorf("cannot get image %s/%s: %w", generateSBOMMessage.ImageNamespace, generateSBOMMessage.ImageName, err)
+	}
+
+	h.logger.Debug("Image found",
+		zap.Any("image", image),
+	)
+
+	sbomFile, err := os.CreateTemp("/tmp", "trivy.sbom.*.json")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary file: %w", err)
+	}
+	defer func() {
+		if err := sbomFile.Close(); err != nil {
+			h.logger.Error("failed to close temporary file", zap.Error(err))
+		}
```

Comment:
nit: I don't think we need to care about closing the file. Do we?

Reply:
this is needed to prevent resource leaks

Reply:
Go has a finalizer, when the GC cleans the sbomFile it will also close the file descriptor. But I think it's nice to be explicit about that, it's more "clean"